cmd/telemetry_provider/telemetry_provider.go

```diff
@@ -0,0 +1,381 @@
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+
+		resp, r, err := authApi.CreateTelemetryProviderConfig().TelemetryProviderSpec(*telemetryProviderSpec).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		telemetryProviderId := resp.GetData().Info.Id
+
+		msg := fmt.Sprintf("The telemetry provider %s is being created", formatter.Colorize(telemetryProviderId, formatter.GREEN_COLOR))
+
+		fmt.Println(msg)
+
+		telemetryProviderCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewTelemetryProviderFormat(viper.GetString("output"), string(resp.GetData().Spec.Type)),
+		}
+
+		respArr := []ybmclient.TelemetryProviderData{resp.GetData()}
+
+		formatter.TelemetryProviderWrite(telemetryProviderCtx, respArr)
+	},
+}
+
+var listTelemetryProviderCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List Telemetry Provider",
+	Long:  "List Telemetry Provider",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+
+		resp, r, err := authApi.ListTelemetryProviders().Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		telemetryProviderCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewTelemetryProviderFormat(viper.GetString("output"), ""),
+		}
+
+		if len(resp.GetData()) < 1 {
+			fmt.Println("No telemetry providers found")
+			return
+		}
+
+		formatter.TelemetryProviderWrite(telemetryProviderCtx, resp.GetData())
+	},
+}
+
+var deleteTelemetryProviderCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete Telemetry Provider",
+	Long:  "Delete Telemetry Provider",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		viper.BindPFlag("force", cmd.Flags().Lookup("force"))
+		configName, _ := cmd.Flags().GetString("config-name")
+		err := util.ConfirmCommand(fmt.Sprintf("Are you sure you want to delete %s: %s", "config", configName), viper.GetBool("force"))
+		if err != nil {
+			logrus.Fatal(err)
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+		configName, _ := cmd.Flags().GetString("config-name")
+
+		config, err := authApi.GetConfigByName(configName)
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		r1, err := authApi.DeleteTelemetryProvider(config.GetInfo().Id).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r1)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		fmt.Printf("Deleting Telemetry Provider %s\n", formatter.Colorize(configName, formatter.GREEN_COLOR))
```

Comment:
Is deleting emitting a task ?

Reply:
yes, create/update/delete all operations create a new task with ybm-backend

Reply:
So in this case usee the wait function. Looks at others like create cluster

Reply:
@shinji62 sorry for the confusion, the telemetry provider apis dont work under a task. So they dont need a wait function.

The DB audit apis which I added do work under a task and needs to be added with a wait method. I will add that in a separate PR.

Reply:
I would love to see that in this PR

Reply:
@shinji62 - This PR is about Telemetry provider lifecycle mgmt. So let's not mix up with the db audit part. 
@aquatiko - For DB Audit part, let's keep the CLI behaviour in sync with what is there for other async operations.

Reply:
@jagdish-yb Ok, so if no task are emitted then that's fine.